typed-racket-lib/typed-racket/utils/any-wrap.rkt

```diff
@@ -105,87 +106,91 @@
  
   (define (any-wrap/traverse v neg-party)
     (define blame+neg-party (cons b neg-party))
-    (match v
-      [(? base-val?)
-       v]
-      [(? unsafe-val?)
-       (fail neg-party v)]
-      [(cons x y) (cons (any-wrap/traverse x neg-party) (any-wrap/traverse y neg-party))]
-      [(? vector? (? immutable?))
-       ;; fixme -- should have an immutable for/vector
-       (vector->immutable-vector
-        (for/vector #:length (vector-length v)
-          ([i (in-vector v)]) (any-wrap/traverse i neg-party)))]
-      [(? box? (? immutable?)) (box-immutable (any-wrap/traverse (unbox v) neg-party))]
-      [(? box?) (chaperone-box v
-                               (lambda (v e)
-                                 (with-contract-continuation-mark
-                                  blame+neg-party
-                                  (any-wrap/traverse e neg-party)))
-                               (lambda (v e) (fail neg-party v)))]
-      ;; fixme -- handling keys properly makes it not a chaperone
-      ;; [(? hasheq? (? immutable?))
-      ;;  (for/hasheq ([(k v) (in-hash v)]) (values k v))]
-      ;; [(? hasheqv? (? immutable?))
-      ;;  (for/hasheqv ([(k v) (in-hash v)]) (values k v))]
-      
-      [(? (λ (e) 
-            (and (hash? e) (immutable? e) 
-                 (not (hash-eqv? e)) (not (hash-eq? e)))))
-       (for/hash ([(k v) (in-hash v)]) (values (any-wrap/traverse k neg-party)
-                                               (any-wrap/traverse v neg-party)))]
-      [(? vector?) (chaperone-vector v
-                                     (lambda (v i e)
-                                       (with-contract-continuation-mark
-                                        blame+neg-party
-                                        (any-wrap/traverse e neg-party)))
-                                     (lambda (v i e) (fail neg-party v)))]
-      [(? hash?) (chaperone-hash v
-                                 (lambda (h k)
-                                   (values k (lambda (h k v)
-                                               (with-contract-continuation-mark
-                                                blame+neg-party
-                                                (any-wrap/traverse v neg-party))))) ;; ref
-                                 (lambda (h k n)
-                                   (if (immutable? v) 
-                                       (values k n)
-                                       (fail neg-party v))) ;; set
-                                 (lambda (h v) v) ;; remove
-                                 (lambda (h k)
+    (parameterize ([current-inspector old-inspector])
```

Comment:
I guess, but then there would have to be two `parameterize` forms, one around the call to the `struct?` predicate, and one around the call to `wrap-struct` in that clause. I thought it would be better to have one `parameterize` over the whole thing instead of that.